### PR TITLE
New Module: `gstama/polyacleanup`

### PR DESCRIPTION
**Pull request by sguizard**
A module for TAMA's tama_flnc_polya_cleanup.py script (https://github.com/GenomeRIK/tama/wiki/TAMA-GO:-Sequence-Cleanup). This script is used to remove remaining polyA tails from Full Length Non Chimeric reads (isoseq 3). 

## PR checklist
- [X] This comment contains a description of changes (with reason).
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/gstama/polyacleanup/main.nf [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("*_tama.fa")                   , emit: fasta
+    tuple val(meta), path("*_tama_polya_flnc_report.txt"), emit: report
+    tuple val(meta), path("*_tama_tails.fa")             , emit: tails
+    path "versions.yml"                                  , emit: versions
+
```
Comment (sateeshperi) [inline modules/gstama/polyacleanup/main.nf:18]:
```suggestion

    when:
    task.ext.when == null || task.ext.when
    
```

Comment (sguizard) [reply modules/gstama/polyacleanup/main.nf:18]: @sateeshperi Thanks for pointing the error. I updated my nf-core and fixed it.

**Comment by sguizard** [comment]
@sateeshperi I re-ran the singularity test and it succeed. The module is ready for review if you wish.

File: modules/gstama/polyacleanup/main.nf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    conda (params.enable_conda ? "bioconda::gs-tama=1.0.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gs-tama:1.0.3--hdfd78af_0':
+        'quay.io/biocontainers/gs-tama:1.0.3--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(fasta)
+
+    output:
+    tuple val(meta), path("*_tama.fa")                   , emit: fasta
```
Comment (sateeshperi) [inline modules/gstama/polyacleanup/main.nf:14]: would it be possible to gzip the output fasta @sguizard

Comment (sguizard) [reply modules/gstama/polyacleanup/main.nf]: I added the output compression on each output. But now, the conda test always fail and docker/singularity tests always succeed. I have the same result locally.

Comment (sateeshperi) [reply modules/gstama/polyacleanup/main.nf]: `/tmp/tmpf9ute79l/conda/env-129b0a0d851494ee9ab3c9b8fdaf71f4/bin/tama_flnc_polya_cleanup.py: Permission denied` 

Im gettting this perm denied running with conda locally

**Review by sateeshperi — APPROVED** [review]
lgtm

File: modules/gstama/polyacleanup/main.nf (hunk 2 above)
Comment (sguizard) [reply modules/gstama/polyacleanup/main.nf]: OK! Thanks for the review!

**Pull request merged**